pkg/sfu/connectionquality/connectionstats.go

```diff
@@ -51,14 +50,59 @@ type ConnectionStats struct {
 func NewConnectionStats(params ConnectionStatsParams) *ConnectionStats {
 	return &ConnectionStats{
 		params:           params,
-		trackSource:      livekit.TrackSource_UNKNOWN,
+		codecName:        getCodecNameFromMime(params.MimeType), // LK-TODO: have to notify on codec change
 		score:            5.0,
 		maxExpectedLayer: buffer.InvalidLayerSpatial,
 		done:             make(chan struct{}),
 	}
 }
 
-func (cs *ConnectionStats) Start() {
+func (cs *ConnectionStats) Start(trackInfo *livekit.TrackInfo) {
+	cs.lock.Lock()
+	cs.trackInfo = trackInfo
+
+	//
+	// get best score and calculate normalization factor.
+	// Raitonale: MOS modeling will yield a max score for a specific codec.
+	// That is outside the control of an SFU. SFU can impair quality due
+	// network issues. So, get the maximum for given codec and normalize
+	// it to the highest score. So, under perfect conditions, it will
+	// yield a MOS of 5 which is the highest rating. Any SFU induced impairment
+	// or network impairment will result in a lower score and that is
+	// the quality that is under SFU/infrastructure control.
+	//
+	if trackInfo.Type == livekit.TrackType_AUDIO {
+		// LK-TODO: would be good to have audio expected bitrate in Trackinfo
+		params := TrackScoreParams{
+			Duration: time.Second,
+			Codec:    cs.codecName,
+			Bytes:    20000 / 8,
+		}
+		cs.normFactors[0] = 5.0 / AudioTrackScore(params, 1)
+	} else {
+		for _, layer := range cs.trackInfo.Layers {
+			spatial := utils.SpatialLayerForQuality(layer.Quality)
+			// LK-TODO: would be good to have expected frame rate in Trackinfo
```

Comment:
Added a couple of bits here that would be good in the protocol (expected frame rate for video layers and expected bit rate for audio)